File: 资本的流通过程/reproduction/simple_reproduction.py
from typing import Dict


def check_simple_equilibrium(
    department_1: Dict[str, float], department_2: Dict[str, float]
) -> bool:
    return department_1["v"] + department_1["m"] == department_2["c"]


if __name__ == "__main__":
    department_1: Dict[str, float] = {"c": 4000.00, "v": 1000.00, "m": 1000.00}
    department_2: Dict[str, float] = {"c": 2000.00, "v": 500.00, "m": 500.00}

    if_simple_equilibrium: bool = check_simple_equilibrium(department_1, department_2)

    if if_simple_equilibrium:
        print("Ⅰv + Ⅰm = Ⅱc")
        print(if_simple_equilibrium)
    else:
        print(if_simple_equilibrium)

File: 资本的流通过程/reproduction/other_code/simple_reproduction.cpp
/* C++：马克思经济学两大部类简单再生产均衡条件的判断*/

#include <array>
#include <iostream>

bool checkSimpleEquilibrium(const std::array<double, 3> &department1,
                            const std::array<double, 3> &department2) {
  return (department1.at(1) + department1.at(2) == department2.at(0)) ? true
                                                                      : false;
}

int main() {
  std::array departmentOne{4000.00, 1000.00, 1000.00};
  std::array departmentTwo{2000.00, 500.00, 500.00};

  bool ifSimpleEquilibrium =
      checkSimpleEquilibrium(departmentOne, departmentTwo);
  std::cout << "Equilibrium of simple reproduction:\n"
            << std::boolalpha << ifSimpleEquilibrium << '\n';
  return 0;
}
